molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/reindex/ReindexActionRegisterService.java

```diff
@@ -0,0 +1,102 @@
+package org.molgenis.data.elasticsearch.reindex;
+
+import static org.molgenis.data.transaction.MolgenisTransactionManager.TRANSACTION_ID_RESOURCE_NAME;
+import static org.molgenis.security.core.runas.RunAsSystemProxy.runAsSystem;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.molgenis.data.DataService;
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.elasticsearch.reindex.ReindexActionMetaData.CudType;
+import org.molgenis.data.elasticsearch.reindex.ReindexActionMetaData.DataType;
+import org.molgenis.data.support.DefaultEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.transaction.support.TransactionSynchronizationManager;
+
+public class ReindexActionRegisterService
+{
+	private static final Logger LOG = LoggerFactory.getLogger(ReindexActionRegisterService.class);
+
+	public static final List<String> EXCLUDED_ENTITIES = Arrays.asList(ReindexActionMetaData.ENTITY_NAME,
+			ReindexActionJobMetaData.ENTITY_NAME);
+
+	private final DataService dataService;
+	private final ReindexActionJobMetaData reindexActionJobMetaData;
+	private final ReindexActionMetaData reindexActionMetaData;
+
+	public ReindexActionRegisterService(DataService dataService, ReindexActionJobMetaData reindexActionJobMetaData,
+			ReindexActionMetaData reindexActionMetaData)
+	{
+		this.dataService = dataService;
+		this.reindexActionJobMetaData = reindexActionJobMetaData;
+		this.reindexActionMetaData = reindexActionMetaData;
+	}
+
+	/**
+	 * Log and create locks for an add/update/delete operation on a Repository
+	 * 
+	 * @param entityMetaData
+	 * @param cudType
+	 */
+	public synchronized void register(EntityMetaData entityMetaData, CudType cudType, DataType dataType, String entityId)
```

Comment:
won't making this method synchronized make the entire dataservice essentially single-threaded?